fuse/data/tokenizers/modular_tokenizer/inject_utils.py

```diff
@@ -209,24 +196,47 @@ def prepare_info_for_model_step(
             elif tokenizer_name.startswith("VECTORS_"):
                 raise NotImplementedError
             else:
-                prev_index_end += len(curr_placeholder_encoding.ids)
-
-        if len(scalars_indices) > 0:
-            scalars_indices = torch.concat(scalars_indices)
-            scalars_values = torch.concat(scalars_values)
-        else:
-            scalars_indices = None
-            scalars_values = None
-
-        if len(scalars_masked_indices) > 0:
-            scalars_masked_indices = torch.tensor(
-                scalars_masked_indices, dtype=torch.int64
+                # prev_index_end += len(curr_placeholder_encoding.ids)
+                curr_scalar_values = torch.full(
+                    (len(curr_placeholder_encoding.ids),),
+                    fill_value=scalar_default_unfound_value,
+                )
+                all_scalars_values.append(curr_scalar_values)
+                all_scalars_valid_mask.append(
+                    torch.full_like(
+                        curr_scalar_values, fill_value=False, dtype=torch.bool
+                    )
+                )
+
+        all_scalars_values = torch.concat(all_scalars_values)
+        all_scalars_valid_mask = torch.concat(all_scalars_valid_mask)
+
+        assert all_scalars_values.shape == all_scalars_valid_mask.shape
+
+        # pad if needed
+        full_query_len = len(token_ids)
+        if full_query_len > all_scalars_values.shape[0]:
```

Comment:
Can't it happend?

Reply:
Do you mean "can it happen" ? 
If so, then yes, almost always.

The main code logic (before it) iterates over each sub part (with specific sub tokenizer) so it does not contain the padding.

I can explain more if it isn't clear